metabase/metabase
# Improve EntityForm component

Preparation to solve #17450

This refactor makes it clearer that the EntityForm component simply wraps or not the actual form in a modal component.

To achieve that, we:

1. remove unused flow types
2. extract the form component call into its own function
3. extract the modal component wrapping into its function
4. simplify ternaries
5. separate definition of complicated props from their passing down the next component

## How to Test

Everything should work as before.

One place this component is used: http://localhost:3000/admin/databases/create
